File: Presentations/WG2_11/GlassBR_Code/fromGOOL/cpp/ReadTable.cpp
#include "ReadTable.hpp"


#include <algorithm>
#include <iostream>
#include <fstream>
#include <iterator>
#include <string>
#include <math.h>
#include <sstream>
#include <vector>

using namespace GlassBR;
using std::string;
using std::vector;
using std::ifstream;
using std::ofstream;

vector<double> GlassBR::read_z_array(string filename) {
    ifstream infile;
    infile.open(filename, std::fstream::in);
    string line;
    std::getline(infile, line);
    infile.close();
    vector<string> z_array_str(0);
    {   std::stringstream ss;
        ss.str(line);
        string word;
        while (std::getline(ss, word, ',')) {
            z_array_str.push_back(word);
        }
    }
    {   vector<string> temp(0);
        for (int i = 1; i < z_array_str.size(); i += 2) {
            temp.push_back(z_array_str.at(i));
        }
        z_array_str = temp;
    }
    vector<double> z_array(0);
    for (int i = 0; i < z_array_str.size(); i++) {
        z_array.push_back(std::stod(z_array_str.at(i)));
    }
    return z_array;
}

vector< vector<double> > GlassBR::read_x_array(string filename) {
    ifstream infile;
    infile.open(filename, std::fstream::in);
    vector<string> lines(0);
    string nextLine;
    while (std::getline(infile, nextLine)) {
        lines.push_back(nextLine);
    }
    infile.close();
    {   vector<string> temp(0);
        for (int i = 1; i < lines.size(); i++) {
            temp.push_back(lines.at(i));
        }
        lines = temp;
    }
    vector< vector<string> > x_array_str(0);
    for (int i = 0; i < lines.size(); i++) {
        vector<string> temp_str(0);
        {   std::stringstream ss;
            ss.str(lines.at(i));
            string word;
            while (std::getline(ss, word, ',')) {
                temp_str.push_back(word);
            }
        }
        {   vector<string> temp(0);
            for (int i = 0; i < temp_str.size(); i += 2) {
                temp.push_back(temp_str.at(i));
            }
            temp_str = temp;
        }
        x_array_str.push_back(temp_str);
    }
    vector< vector<double> > x_array(0);
    for (int i = 0; i < x_array_str.size(); i++) {
        vector<double> nextLine(0);
        for (int j = 0; j < x_array_str.at(i).size(); j++) {
            nextLine.push_back(std::stod(x_array_str.at(i).at(j)));
        }
        x_array.push_back(nextLine);
    }
    return x_array;
}

vector< vector<double> > GlassBR::read_y_array(string filename) {
    ifstream infile;
    infile.open(filename, std::fstream::in);
    vector<string> lines(0);
    string nextLine;
    while (std::getline(infile, nextLine)) {
        lines.push_back(nextLine);
    }
    infile.close();
    {   vector<string> temp(0);
        for (int i = 1; i < lines.size(); i++) {
            temp.push_back(lines.at(i));
        }
        lines = temp;
    }
    vector< vector<string> > y_array_str(0);
    for (int i = 0; i < lines.size(); i++) {
        vector<string> temp_str(0);
        {   std::stringstream ss;
            ss.str(lines.at(i));
            string word;
            while (std::getline(ss, word, ',')) {
                temp_str.push_back(word);
            }
        }
        {   vector<string> temp(0);
            for (int i = 1; i < temp_str.size(); i += 2) {
                temp.push_back(temp_str.at(i));
            }
            temp_str = temp;
        }
        y_array_str.push_back(temp_str);
    }
    vector< vector<double> > y_array(0);
    for (int i = 0; i < y_array_str.size(); i++) {
        vector<double> nextLine(0);
        for (int j = 0; j < y_array_str.at(i).size(); j++) {
            nextLine.push_back(std::stod(y_array_str.at(i).at(j)));
        }
        y_array.push_back(nextLine);
    }
    return y_array;
}

File: Presentations/WG2_11/GlassBR_Code/fromGOOL/python/ReadTable.py
from __future__ import print_function
import sys
import math


def read_z_array(filename):
    infile = open(filename, "r")
    line = infile.readline()
    infile.close()
    z_array_str = []
    z_array_str = line.split(",")
    z_array_str = z_array_str[1::2]
    z_array = []
    for i in range(len(z_array_str)):
        z_array.append(float(z_array_str[i]))
    return z_array

def read_x_array(filename):
    infile = open(filename, "r")
    lines = []
    lines = infile.readlines()
    infile.close()
    lines = lines[1::]
    x_array_str = []
    for i in range(len(lines)):
        temp_str = []
        temp_str = lines[i].split(",")
        temp_str = temp_str[0::2]
        x_array_str.append(temp_str)
    x_array = []
    for i in range(len(x_array_str)):
        nextLine = []
        for j in range(len(x_array_str[i])):
            nextLine.append(float(x_array_str[i][j]))
        x_array.append(nextLine)
    return x_array

def read_y_array(filename):
    infile = open(filename, "r")
    lines = []
    lines = infile.readlines()
    infile.close()
    lines = lines[1::]
    y_array_str = []
    for i in range(len(lines)):
        temp_str = []
        temp_str = lines[i].split(",")
        temp_str = temp_str[1::2]
        y_array_str.append(temp_str)
    y_array = []
    for i in range(len(y_array_str)):
        nextLine = []
        for j in range(len(y_array_str[i])):
            nextLine.append(float(y_array_str[i][j]))
        y_array.append(nextLine)
    return y_array


